src/server/scripts/Zandalar/AtalDazar/boss_volkaal.cpp

```diff
@@ -0,0 +1,420 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTrigger.h"
+#include "AreaTriggerAI.h"
+#include "CellImpl.h"
+#include "Creature.h"
+#include "GridNotifiersImpl.h"
+#include "InstanceScript.h"
+#include "ObjectAccessor.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "SpellAuraEffects.h"
+#include "SpellScript.h"
+#include "TemporarySummon.h"
+#include "atal_dazar.h"
+
+enum VolkaalSpells
+{
+    SPELL_SOUL_ANCHOR         = 259537,
+    SPELL_BAD_VOODOO          = 250192,
+    SPELL_RAPID_DECAY         = 250241,
+    SPELL_RAPID_DECAY_TARGET  = 250694,
+    SPELL_RAPID_DECAY_RANDOM  = 250697,
+    SPELL_TOXIC_POOL          = 250585,
+    SPELL_REANIMATE           = 259531,
+    SPELL_TOXIC_LEAP_SELECTOR = 250708,
+    SPELL_TOXIC_LEAP          = 250258,
+    SPELL_TOXIC_LEAP_DAMAGE   = 250259,
+    SPELL_NOXIOUS_STENCH      = 259572,
+    SPELL_LINGERING_NAUSEA    = 250372
+};
+
+enum VolkaalEvents
+{
+    EVENT_TOXIC_LEAP = 1,
+    EVENT_NOXIOUS_STENCH,
+};
+
+enum VolkaalActions
+{
+    ACTION_TOTEMS_DIED = 1
+};
+
+enum VolkaalSummonGroups
+{
+    SUMMON_GROUP_VOLKAAL_TOTEMS = 0
+};
+
+enum VolkaalTexts
+{
+    SAY_AGGRO = 0,
+    SAY_DECAY = 1,
+    SAY_DIED  = 2,
+    SAY_SLAY  = 3,
+    SAY_WIPE  = 4
+};
+
+// 122965 - Vol'kaal
+struct boss_volkaal : public BossAI
+{
+    boss_volkaal(Creature* creature) : BossAI(creature, DATA_VOLKAAL) { }
+
+    void Reset() override
+    {
+        _Reset();
+    }
+
+    void JustAppeared() override
```

Comment:
Maybe moving JustAppeared at the top since it's the first thing that gets called?